File: backend/src/main/java/com/backend/petplace/domain/place/dto/KcisaDto.java
package com.backend.petplace.domain.place.dto;

import java.util.List;

public record KcisaDto(Response response) {

  public record Response(Header header, Body body) {}

  public record Header(String resultCode, String resultMsg) {}

  public record Body(Items items, String numOfRows, String pageNo, String totalCount) {}

  public record Items(List<Item> item) {}

  public record Item(
      String title,
      String issuedDate,
      String category1,
      String category2,
      String category3,
      String description,
      String tel,
      String url,
      String address,
      String coordinates,
      String charge
  ) {}
}

File: backend/src/main/java/com/backend/petplace/domain/place/dto/KcisaDto.kt
package com.backend.petplace.domain.place.dto

data class KcisaDto(
    val response: Response?
) {
    data class Response(
        val header: Header?,
        val body: Body?
    )

    data class Header(
        val resultCode: String?,
        val resultMsg: String?
    )

    data class Body(
        val items: Items?,
        val numOfRows: String?,
        val pageNo: String?,
        val totalCount: String?
    )

    data class Items(
        val item: List<Item>?
    )

    data class Item(
        val title: String?,
        val issuedDate: String?,
        val category1: String?,
        val category2: String?,
        val category3: String?,
        val description: String?,
        val tel: String?,
        val url: String?,
        val address: String?,
        val coordinates: String?,
        val charge: String?
    )
}
